GraphQLParser.cpp

```diff
@@ -20,36 +20,56 @@ namespace graphql {
 
 // Given properly-configured yylex, run the parser and return the
 // result.
-static std::unique_ptr<ast::Node> doParse(const char **outError, yyscan_t scanner) {
+static std::unique_ptr<ast::Node> doParse(const char **outError, yyscan_t scanner, bool enableSchema) {
   Node *outAST;
-  yy::GraphQLParserImpl parser(&outAST, outError, scanner);
+  yy::GraphQLParserImpl parser(enableSchema, &outAST, outError, scanner);
   int failure = parser.parse();
   return !failure ? std::unique_ptr<ast::Node>(outAST) : nullptr;
 }
 
-std::unique_ptr<ast::Node> parseString(const char *text, const char **error) {
+static std::unique_ptr<ast::Node> parseStringImpl(const char *text, const char **error, bool enableSchema) {
   yyscan_t scanner;
   struct LexerExtra extra;
   yylex_init_extra(&extra, &scanner);
   YY_BUFFER_STATE buffer = yy_scan_string(text, scanner);
   yy_switch_to_buffer(buffer, scanner);
 
-  auto result = doParse(error, scanner);
+  auto result = doParse(error, scanner, enableSchema);
   yylex_destroy(scanner);
   return result;
 }
 
-std::unique_ptr<ast::Node> parseFile(FILE *file, const char **error) {
+std::unique_ptr<ast::Node> parseString(const char *text, const char **error) {
+  return parseStringImpl(text, error, false);
```

Comment:
nit: it's hard to guess what is the meaning of the boolean without reading a type signature of parseStringImpl, so it's usually a good practice to leave a comment `/* enableSchema */` right next to a parameter.